File: 01-presentation-blog-examples/src/main/java/com/nalepa/demo/example03/Example03_Blocking_IO.java
package com.nalepa.demo.example03;

import java.time.LocalTime;

public class Example03_Blocking_IO {
    public static void main(String[] args) {
        simulateBlockingIO();
    }

    static void simulateBlockingIO() {
        log("Start blocking IO code");
        // http call, database call, reading from file, whatever what needs external data
        nonThrowingSleep(5000);
        log("End blocking IO code");
    }

    static void log(String message) {
        System.out.println(Thread.currentThread().getName() + " : " + LocalTime.now() + " : " + message);
    }

    static void nonThrowingSleep(long millis) {
        try {
            Thread.sleep(millis);
        } catch (InterruptedException e) {
            throw new RuntimeException(e);
        }
    }
}

File: 01-presentation-blog-examples/src/main/kotlin/com/nalepa/demo/example03/Example03_Blocking_IO.kt
package com.nalepa.demo.example03

import java.time.LocalTime

fun main() {
    simulateBlockingIO()
}

fun simulateBlockingIO() {
    log("Start blocking IO code")
    Thread.sleep(5000) // http call, database call, reading from file, whatever what needs external data
    log("End blocking IO code")
}

fun log(message: String) {
    println("${Thread.currentThread().name} : ${LocalTime.now()} : $message")
}
